### ISSUE TITLE
[GR-57593] Implement Exception Handling Proposal for WebAssembly

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9764]
### TL;DR

We plan to implement [the Exception Handling proposal](https://github.com/WebAssembly/exception-handling) in GraalWasm.

### Details

This proposal lets WebAssembly modules throw and catch exceptions.
WebAssembly modules can define exception types, called tags, which have typed arguments (fields of the exception record). Code in WebAssembly will be able to throw an exception having a given tag and given arguments. Such an exception can then be caught in the embedder, be it Java or another Graal Language like JavaScript. Once caught, the tag and arguments of the exception can be retrieved and inspected. WebAssemlby code will also be able to catch exceptions, be it exceptions that originate in WebAssembly, in Java, or other Graal Languages.

More details at https://github.com/WebAssembly/exception-handling/blob/main/proposals/exception-handling/Exceptions.md
